EMF2014/Deserializer.cpp

```diff
@@ -32,21 +26,24 @@
  SOFTWARE.
  */
 
-#ifndef _OUTGOING_MESSAGES_MANAGER_H_
-#define _OUTGOING_MESSAGES_MANAGER_H_
-
-#include <Arduino.h>
-#include <FreeRTOS_ARM.h>
-#include "EMF2014Config.h"
+#include "Deserializer.h"
+#include "Serializable.h"
+#include "Utils.h"
+#include "Tilda.h"
 
-class OutgoingMessagesManager {
-public:
-    OutgoingMessagesManager();
+#include "YoApp.h"
 
-    /* This is a blocking method */
-    void handleTransmissionWindow(TickType_t duration);
-private:
+Serializable* Deserializer::deserialize(const byte* data) {
+    uint16_t messageTypeId = data[1] * 256 + data[0];
 
-};
+    if (messageTypeId == MESSAGE_TYPE_ID_YO) {
```

Comment:
I think we had something similar to what I'd want before. Would be better if when an app started it said "hey, I can handle a particular message id", then the deserializer only has the responsibility to desearailze the packet header and decide who to send the rest of the message from its message handler list. This also has the benefit (other than not having to alter this class for every new message type) that we only have to deserialse messages for running apps.